application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,319 @@
+package org.togetherjava.tjbot.commands.basic;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.MessageBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.events.interaction.SelectionMenuEvent;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.interactions.commands.build.SubcommandData;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.ComponentInteraction;
+import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
+import net.dv8tion.jda.api.interactions.components.selections.SelectionMenu;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * Implements the {@code roleSelect} command.
+ *
+ * <p>
+ * Allows users to select their roles without using reactions, instead it uses selection menus where
+ * you can select multiple roles. <br />
+ * Note: the bot can only use roles below its highest one
```

Comment:
This could be slightly rephrased to be a bit more easy to understand.
Example: "Note: The bot can only use roles with a permission level below it's highest one"

Not sure if the example given is a suitable substitution 😆

Reply:
While I get the point, don't like the way that phrases it either.

It can only use roles with a lower position, not less permissions,

Reply:
You're right. I miss-worded that really badly, but the same logic applies, if you agree.
Something like: `Note: the bot can only use roles with a position below its highest one`
Just adding the "with a position" makes it a bit easier to understand, I believe

Reply:
```suggestion
* Note: the bot can only use roles with a position below its highest one
```
I fully agree!